### PR TITLE
File upload post auth and pdf upload 

### PR DESCRIPTION
**Pull request by kayladavis**
This pr covers authing file posts (making sure the user is a student or admin) and also pdf upload.

### CODE REVIEW

**Comment by kayladavis** [comment]
also connects to #129, but zenhub doesn't seem to allow attaching more than one issue

File: backend/server.js [hunk 1]
```diff
@@ -11,7 +11,9 @@ import models from './models'
 import router from './routes'
 import schema from './schema'
 import parseJwtUser from './middleware/parseJwtUser'
-import imageUploader from './upload'
+import { imageUploader, pdfUploader } from './upload'
+import { allowedToSubmit } from './helpers/submission'
```
Comment (robmcl4) [inline backend/server.js]: seems like an unused import

File: backend/test/routes/fileUpload.js [hunk 2]
```diff
@@ -0,0 +1,126 @@
+import request from 'supertest'
```
Comment (robmcl4) [inline backend/test/routes/fileUpload.js]: Please also re-format to 2 spaces

File: backend/upload.js [hunk 3]
```diff
@@ -4,46 +4,75 @@ import uuidv1 from 'uuid/v1'
...
-export default function uploader (req, res) {
-  upload(req, res, (err) => {
-    if (err) {
-      if(err.message === 'File too large'){
-        return res.status(413).json({error: 'File too large'})
-      }
-      return res.status(500).json({error: 'Failed to Upload'})
-    }
+function pdfTypeFilter (req, file, cb, mimetype) {
+  if (file.mimetype !== 'application/pdf') { // verify image is a jpg
```
Comment (robmcl4) [inline backend/upload.js]: jpg -> pdf

File: backend/upload.js [hunk 4]
```diff
@@ -4,46 +4,75 @@ import uuidv1 from 'uuid/v1'
 import nconf from './config'
```
Comment (robmcl4) [inline backend/upload.js]: I think this file should be moved to `backend/routes/upload.js`, which allows us to use the Router module in combination with a small edit of `backend/routes/index.js`. As a bonus, we can leave `backend/server.js` fairly pristine.

Router is pretty clean to use:

```javascript
const router = Router()

router.route('/path/to/thing')
  .get(
    middlewareFunc, // probably your uploadAuth function goes here
    (req, res, next) => imageUpload( /* ... */ )
  )

export default router
```

File: backend/server.js [hunk 5]
```diff
@@ -34,7 +36,19 @@ if (config.get('NODE_ENV') !== 'production') {
   app.get('/graphiql', graphiqlExpress({ endpointURL: '/graphql' }))
 }
 
-app.use('/static', express.static(path.join(__dirname, 'images')))
-router.post('/static/upload', imageUploader) // TODO: Require Authentication
+function uploadAuth(req, res, next) {
```
Comment (robmcl4) [inline backend/server.js]: this middleware function feels odd defined here, maybe move it to `backend/middleware` or, if you're changing the `upload.js` file to use Router (see other comment), into there?

**Pull request merged**